src/main/java/trackr/model/order/OrderStatus.java

```diff
@@ -0,0 +1,80 @@
+package trackr.model.order;
+
+import static java.util.Objects.requireNonNull;
+import static trackr.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an Order's status in the order list.
+ * Guaruntees: immutable; is valid as declared in {@link #isValidOrdeStatus(String)}
+ */
+public class OrderStatus {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Order status should only be `N` or `n` for Not Delivered, `I` or `i` for In Progress,"
+            + " or `D` or `d` for Delivered";
```

Comment:
Since simplified to 3 status states instead of ones proposed, maybe the UG can be updated accordingly